msr-flows/msbr_compare/run_user_input.py

```diff
@@ -341,159 +433,378 @@ def cycle_time_decay(self, identifier='CTD'):
 
 if __name__ == '__main__':
 
-    output_path = f'./{ui.path_to_dump_files}/'
-    misc_funcs.set_directory(output_path)
-    close_boolean = False
+    base_output_path = f'./{ui.path_to_dump_files}/'
+    misc_funcs.set_directory(base_output_path)
+
     element_dictionary = dict()
+    active_identifiers = list()
 
     for index in range(len(ui.element_flow_list)):
         element_dictionary[ui.element_flow_list[index]] = [
             ui.associated_symbol_list[index], ui.associated_atomic_list[index]]
+    for N_index, N_steps in enumerate(ui.number_serp_steps_list):
```

Comment:
Alot of the code in this very big for loop can be easily functionalized

Reply:
I moved the active run list creation to the `user_input` file to be read by `run_user_input`. I then created a function to perform the full task and then ran each for the function. This is a much better approach, I'm glad you pointed this out.